misc/services/mobile-webui/mobile-webui-frontend/src/reducers/wfProcesses_status/picking.js

```diff
@@ -148,12 +284,89 @@ const normalizePickingLines = (lines) => {
       ...line,
       steps: line.steps.reduce((accum, step) => {
         accum[step.pickingStepId] = step;
+        accum[step.pickingStepId].altSteps = { qtyToPick: 0, genSteps: {} };
+
+        // Mock generated steps - used for testing w/o real data
+        // accum[step.pickingStepId].altSteps.genSteps = {
+        //   1000819: {
+        //     id: '1000819',
+        //     locatorName: 'Hauptlager',
+        //     huBarcode: '1000437',
+        //     uom: 'Kg',
+        //     qtyAvailable: 45,
+        //     qtyPicked: 0,
+        //   },
+        //   1000820: {
+        //     id: '1000820',
+        //     locatorName: 'Hauptlager',
+        //     huBarcode: '1000463',
+        //     uom: 'Kg',
+        //     qtyAvailable: 25,
+        //     qtyPicked: 0,
+        //   },
+        // };
+
         return accum;
       }, {}),
     };
   });
 };
 
+export const mergeActivityDataStoredAndGenerateAltSteps = ({ draftActivityDataStored, fromActivity }) => {
+  const { lines } = fromActivity.componentProps;
+  let genSteps = {};
+
+  // loop within steps
+  for (let lineIdx = 0; lineIdx < lines.length; lineIdx++) {
+    // console.log('ABOUT_TO_SEND:', draftActivityDataStored.dataStored.lines[lineIdx]);
+    computeLineStatus({ draftLine: draftActivityDataStored.dataStored.lines[lineIdx] });
+
+    for (let stepIdx = 0; stepIdx < lines[lineIdx].steps.length; stepIdx++) {
+      let step = lines[lineIdx].steps[stepIdx];
+      let { qtyRejected } = lines[lineIdx].steps[stepIdx].mainPickFrom;
+      let totalAltQtys = 0;
+
+      // allocateQtys in sync with the data received from the BE
+      let pickFromAlternatives = fromActivity.componentProps.pickFromAlternatives;
+      for (let altKey in pickFromAlternatives) {
+        if (
+          step.pickFromAlternatives[pickFromAlternatives[altKey].id] &&
+          step.pickFromAlternatives[pickFromAlternatives[altKey].id].qtyPicked > 0
+        ) {
+          totalAltQtys = totalAltQtys + step.pickFromAlternatives[pickFromAlternatives[altKey].id].qtyPicked;
+
+          genSteps[pickFromAlternatives[altKey].id] = pickFromAlternatives[altKey];
+          genSteps[pickFromAlternatives[altKey].id].qtyPicked =
```

Comment:
can you pls extract those long things like 
`lines[lineIdx].steps[stepIdx]`
`step.pickFromAlternatives[pickFromAlternatives[altKey].id]`
`pickFromAlternatives[altKey]`
etc etc

into some intermediate variables with relevant names.

in this shape, for me is dramatically hard to follow.